Display zero value when using shipping classes

Chunkford opened the pull request:
When offering free shipping on certain products via a shipping class, the price (albeit zero value) isn't shown to the end user.
![2018-01-27_1320](https://user-images.githubusercontent.com/11464425/35472360-9dd2ee94-0365-11e8-8ca9-3aad6cbb770b.png)


It makes for a better user experience to actually show the 0.00 price to ensure there isn't any doubt and stop any abandoned carts or useless customer interaction
![2018-01-27_1321](https://user-images.githubusercontent.com/11464425/35472363-a185414a-0365-11e8-91b5-8392e433f754.png)

Chunkford committed:
Display zero value when using shipping classes

When offering free shipping on certain products via a shipping class, the price (albeit zero value) isn't shown to the end user.
https://www.screencast.com/t/8oT7maIftTZo

It makes for a better user experience to actually show the 0.00 price to ensure there isn't any doubt.
https://www.screencast.com/t/mkrKWNItrI5r

WPprodigy commented:
Mmm, I recall this being removed on purpose at some point in time. Had to do with Free Shipping I believe. For example: "Free Shipping: $0.00" is kinda silly.

claudiosanches commented:
Not sure anymore in what version we removed it.
We could hide only for free shipping, or close it and let people use `woocommerce_cart_shipping_method_full_label` to crate custom labels.

WPprodigy commented:
Well, as-is, you could just rename the rate with the title field, right? So `2nd Class: $0.00` would be the rate name.

Chunkford committed:
Update wc-cart-functions.php

Alteration to the proposed changes to take into account free shipping so the price isn't displayed

Chunkford commented:
2nd Class @ £0.00 is only free for a certain product. Anything else added to the cart then the 2nd class shipping is £2.99 as per the calculation type set to per order. Changing the rate name wouldn't work in this scenario.

The purpose of the suggest change was all it needed was an `=` for a solution rather than use the `woocommerce_cart_shipping_method_full_label` hook, a solution which has brought around other scenarios I didn't think about.

I've made an alteration to the proposed change. Is this better suited for the other situations not thought off?

Chunkford committed:
Update wc-cart-functions.php

claudiosanches commented:
Need to use `===`, but I like this change.
Makes sense in some scenarios to display `0.00`.
cc @mikejolley

Chunkford committed:
Update wc-cart-functions.php

Chunkford commented:
@claudiosanches I presume you mean `!==` not `===`?
I've made the change based on the former
